eng/common/scripts/Service-Level-Readme-Automation.ps1

```diff
@@ -87,16 +90,22 @@ function update-metadata-table($readmeFolder, $readmeName, $serviceName, $msServ
 {
   $readmePath = Join-Path $readmeFolder -ChildPath $readmeName
   $readmeContent = Get-Content -Path $readmePath -Raw
-  $null = $readmeContent -match "---`n*(?<metadata>(.*`n)*)---`n*(?<content>(.*`n)*)"
-  $restContent = $Matches["content"]
-  $orignalMetadata = $Matches["metadata"]
   # $Language, $LanguageDisplayName are the variables globally defined in Language-Settings.ps1
   $metadataString = GenerateDocsMsMetadata -language $Language -languageDisplayName $LanguageDisplayName -serviceName $serviceName `
     -tenantId $TenantId -clientId $ClientId -clientSecret $ClientSecret `
     -msService $msService
-  $null = $metadataString -match "---`n*(?<metadata>(.*`n)*)---"
-  $mergedMetadata = compare-and-merge-metadata -original $orignalMetadata -updated $Matches["metadata"]
-  Set-Content -Path $readmePath -Value "---`n$mergedMetadata---`n$restContent" -NoNewline
+  $match = $readmeContent -match "^---\n*(?<metadata>(.*\n?)*?)---\n*(?<content>(.*\n?)*)"
+  if (!$match) {
+    Set-Content -Path $readmePath -Value "$metadataString`r`n$readmeContent" -NoNewline
+    return
+  }
+  $restContent = $Matches["content"]
+  $metadata = $Matches["metadata"]
+  $metadataMatch = $metadataString -match "---\n(?<metadata>(.*\n)*)---"
```

Comment:
While this works something to consider for potential future improvements.
1) We should consider passing in the original metadata into the GenerateDocsMsMetadata function and let it generate and merge the metadata. That way we don't have to reparse the metadata string and then do the merging and then redo the `---` tags around it when we write. 
2) If we were to keep this extra parsing we should split the regex up so we can share it with the other place above that is also parsing the metadata so we can ensure they remain consistent.

Reply:
I have moved the logic into generateDocsMsMetadata.